src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java

```diff
@@ -387,7 +388,7 @@ public CatalogRawObject getCatalogObjectRevisionRaw(String bucketName, String na
     }
 
     public CatalogObjectMetadata restore(String bucketName, String name, Long commitTime) {
-        CatalogObjectRevisionEntity catalogObjectRevision = catalogObjectRevisionRepository.findCatalogObjectRevisionByCommitTime(bucketName,
+        CatalogObjectRevisionEntity catalogObjectRevision = catalogObjectRevisionRepository.findCatalogObjectRevisionByCommitTime(Arrays.asList(bucketName),
```

Comment:
The called method should not do any changes to your structure? So `Collections.singletonList` might be more appropriate here